java/sage/MediaServerRemuxer.java

```diff
@@ -884,9 +887,9 @@ private void pushData(byte[] buffer, int offset, int length)
       if (buffer[offset] != 0x47 || !tsSynced)
       {
         int startingOffset = offset;
-        if (Sage.DBG) System.out.println("Remuxer is out of sync.");
+        if (Sage.DBG && tsSynced) System.out.println("Remuxer is out of sync.");
 
-        while (length - (offset + 377) > 0)
+        while (length - (offset + (188 * 2)) > 0)
```

Comment:
You're correct. The offset could be 1 over 188 * 2 and that would cause an out of bounds exception. Due to the alignment buffering, the code mentioned here doesn't get hit very often, so it was easy to miss. While I'm not a Java veteran, small mistakes like that are quite embarrassing.